atoma-state/src/metrics.rs

```diff
@@ -0,0 +1,707 @@
+use std::collections::HashMap;
+
+use atoma_p2p::metrics::{
+    ChatCompletionsMetrics, EmbeddingsMetrics, ImageGenerationMetrics, ModelMetrics, NodeMetrics,
+};
+use once_cell::sync::Lazy;
+use prometheus::{GaugeVec, Opts, Registry};
+use serde::Deserialize;
+use tokio::{
+    sync::{mpsc, watch},
+    task::JoinHandle,
+};
+use tracing::instrument;
+
+use crate::{
+    errors::MetricsServiceError,
+    timer::{trigger_new_metrics_collection_task, Modalities},
+};
+
+/// Metrics timeout, in seconds
+const METRICS_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(1);
```

Comment:
Let's make this configurable

Reply:
What's the benefit ?

Reply:
The main benefit is that we may notice that metrics are slower in being scraped that first anticipated, and rather than having to do a programmatic change we can change a config and re-deploy.

Reply:
That is true, but ideally any deployed proxy will follow the default value, unless having a very good reason to not do so. But I can add it to the config